xwiki-platform-core/xwiki-platform-menu/xwiki-platform-menu-ui/src/main/resources/Menu/MenuMacro.xml

```diff
@@ -189,15 +188,13 @@ require(['jquery'], function($) {
 
   function setDropdownButtonTitle(dropDownButton) {
     var xDropdown = $(dropDownButton).parent().parent();
-    require(['xwiki-l10n!menu-ui-translation-keys'], function(l10n) {
-      if($(dropDownButton).parent().parent().hasClass('open')){
-        $(dropDownButton).attr('title',l10n['closeSubMenu']);
-        $(xDropdown).attr('aria-expanded', "true");
-      }else{
-        $(dropDownButton).attr('title',l10n['openSubMenu']);
-        $(xDropdown).attr('aria-expanded', "false");
-      }
-    });
+    if($(dropDownButton).parent().parent().hasClass('open')){
+      $(dropDownButton).attr('title',l10n['closeSubMenu']);
+      $(xDropdown).attr('aria-expanded', "true");
+    }else{
```

Comment:
```suggestion
    } else {
```

Reply:
Is there a way to check formatting in xml files automatically or is it just a matter of code relecture and good practices ?

Reply:
Several options:
- Format by hand (very painful)
- You can copy and past in an IntelliJ (IJ) scratch file (or vscode) and format there before copying back (painful)
- Use the Language Injection feature of IJ (https://www.jetbrains.com/help/idea/using-language-injections.html#edit_injected_fragment). That way the content of an xml element is analyzed as a specific language inside the xml element (just avoid the "Open an injection editor﻿" method, it will make a mess)

Reply:
You can teach IntelliJ to automatically interpret the content of certain XML elements as a certain language (JavaScript support requires the Ultimate Edition, though), see the [documentation how to configure language injections](https://www.jetbrains.com/help/idea/using-language-injections.html#configure-injection-rules). I've configured it to interpret the content of a tag with name `code` and XPath condition `../../className/text() = "XWiki.JavaScriptExtension"` as JavaScript. This gives me the full JavaScript formatting and analysis support in IntelliJ when looking at an XWiki document. To be honest, I'm not sure though the IntelliJ configuration we provide includes formatting for JavaScript as when looking at this code in IntelliJ I don't see any formatting complaints (or this is just not working in the injected field?). But anyway, language injections are great and make editing XWiki documents so much nicer that I frequently just edit the document in IntelliJ and just import it in XWiki for testing.

Reply:
Also, @raphj developed a _prototype_ to automatically synchronize an XWiki space with the source code of an `-ui` module.
Avoiding the step where you have to manually copy the javascript to the wiki.